donkeycar/parts/actuator.py

```diff
@@ -81,25 +208,29 @@ def set_pulse(self, pulse):
     def run(self, pulse):
         self.set_pulse(pulse)
 
+
 class PWMSteering:
     """
-    Wrapper over a PWM motor controller to convert angles to PWM pulses.
+    Wrapper over a PWM pulse controller to convert angles to PWM pulses.
     """
     LEFT_ANGLE = -1
     RIGHT_ANGLE = 1
 
-    def __init__(self,
-                 controller=None,
-                 left_pulse=290,
-                 right_pulse=490):
+    def __init__(self, controller, left_pulse, right_pulse):
+
+        if controller is None:
+            raise ValueError("PWMSteering requires a set_pulse controller to be passed")
+        set_pulse = getattr(controller, "set_pulse", None)
+        if set_pulse is None or not callable(set_pulse):
+            raise ValueError("controller must have a set_pulse method")
 
         self.controller = controller
         self.left_pulse = left_pulse
         self.right_pulse = right_pulse
         self.pulse = dk.utils.map_range(0, self.LEFT_ANGLE, self.RIGHT_ANGLE,
                                         self.left_pulse, self.right_pulse)
         self.running = True
-        print('PWM Steering created')
+        logger.debug('PWM Steering created')
```

Comment:
Just an observation, I think we have statements like these constructor prints on the `info` level. Such initialisations are printed only once in the car app, so it doesn't clog up the shell anyway. Debug output would be for printing some data within the `run()` method or other detailed information that you really don't want to see, if you don't debug.

Reply:
Good point.  I switched the logging calls in the constructors to be info level.